apps/typeform/frontend/src/AppConfig/AppConfig.tsx

```diff
@@ -0,0 +1,231 @@
+    } catch (error) {
+      this.props.sdk.notifier.error(error);
+    }
+  };
+
+  normalizeWorkspaceResponse = (response: WorkspacesResponse) => {
+    return response.workspaces.items.map(workspace => ({
+      name: workspace.name,
+      id: workspace.id
+    }));
+  };
+
+  onAppConfigure = () => {
+    const { accessToken, workspaceId, contentTypes, selectedFields } = this.state;
+    const parameters = { accessToken, workspaceId };
+    const error = validateParameters(parameters);
+    const hasStaleWorkspaceIdSelected = !this.selectedWorkspaceIdIsValid();
+
+    if (error) {
+      this.props.sdk.notifier.error(error);
+      return false;
+    }
+
+    if (hasStaleWorkspaceIdSelected) {
+      this.props.sdk.notifier.error('Select a valid workspace.');
+      return false;
+    }
+
+    return {
+      parameters: { accessToken, workspaceId },
+      targetState: selectedFieldsToTargetState(contentTypes, selectedFields)
+    };
+  };
+
+  selectedWorkspaceIdIsValid = (): boolean => {
+    return !!this.state.workspaces.find(workspace => workspace.id === this.state.workspaceId);
+  };
+
+  setWorkSpaceId = (id: string) => {
+    this.setState({ workspaceId: id.trim() });
+  };
+
+  setAccessToken = (token: string) => {
+    this.setState({ accessToken: token.trim() });
+  };
+
+  onSelectedFieldsChange = (selectedFields: SelectedFields) => {
+    this.setState({ selectedFields });
+  };
+
+  render() {
+    const { contentTypes, compatibleFields, selectedFields } = this.state;
+
+    return (
+      <div>
+        <div className={styles.background('#262627')} />
+        <div className={styles.body}>
+          <div>
+            <div>
+              {}
+              <Typography>
+                <Heading>About Typeform</Heading>
+                <Paragraph className={styles.aboutP}>
+                  The{' '}
+                  <TextLink
+                    href="https://www.typeform.com/"
+                    target="_blank"
+                    rel="noopener noreferrer">
+                    Typeform
+                  </TextLink>{' '}
+                  app allows you to reference your forms from Typeform without leaving Contentful.
+                </Paragraph>
+              </Typography>
+              <hr className={styles.splitter} />
+            </div>
+            <div>
+              <Typography>
+                <Heading>Configuration</Heading>
+                <FormLabel htmlFor="workspaceId" required>
+                  Select the Typeform workspace you want to connect
+                </FormLabel>
+                <Select
+                  id="workspaceId"
+                  name="workspaceId"
+                  onChange={(event: any) => this.setWorkSpaceId(event.currentTarget.value)}
+                  hasError={this.state.workspaces.length > 0 && !this.selectedWorkspaceIdIsValid()}
+                  value={this.state.workspaceId}
+                  data-test-id="typeform-select">
```

Comment:
Should we add `disabled={this.state.workspaces.length === 0}`?

Reply:
We can add that for the Typeform field but workspaces can not be empty since Typeform enforces you to have at least one workspace

Reply:
Seems unnecessary then.